s11d: Decompile rope and dsmoke2 systems

mak1jk opened the pull request:
This PR decompiles the rope and dsmoke2 systems for the s11d overlay.

## Changes
- **rope.c/h**: Full rope system implementation
- **dsmoke2.c/h**: Smoke effect system implementation


## Notes
- Split from PR #2135 as requested by @KieronJ
- Contains only working, buildable code (no `#if 0` guards)
- WIP functions (hind_bul, landing) will remain in #2135 for future work

Refs #2135

mak1jk committed:
s11d: decompile rope and dsmoke2 systems

- Add rope.c and rope.h with full rope system implementation
- Add dsmoke2.c and dsmoke2.h with smoke effect system
- All functions build cleanly and match original binaries
- Split from PR #2135 as requested by @KieronJ

mak1jk commented:
i can't find all of the names of  func_Address in the funtions.txt do you have a script for that or should i make my own or am i missing somenthing?

mak1jk commented:
ok I'm looking closer to here:"python decompme_asm.py [path to .s file] to have the assembly instructions"

GirianSeed commented on source/overlays/s11d/chara/rope/rope.h:
```diff
@@ -0,0 +1,143 @@
+extern void *data_800BBA20;
+extern int data_800BBA40;
+extern int data_800D1F84;
+extern int func_80031298(void *);
```
The address in ``func_80031298`` appears to be bogus. According to the linker map, that places in in the middle of GM_SetMap.

mak1jk committed:
s11d: Fix symbol naming and add INCLUDE_ASM for dsmoke2

Address reviewer feedback from PR #2139:

Symbol Naming Fixes:
- Rename data_800D1F84 → s11d_dword_800D1F84 (overlay-specific symbol)
- Rename data_800BB9B4 → gControlCount_800AB9B4 (correct main exe symbol)

Code Organization:
- Add INCLUDE_ASM pragmas for unimplemented dsmoke2 functions:
  * s11d_dsmoke2_800CC0E0
  * s11d_dsmoke2_800CC1DC
  * s11d_dsmoke2_800CC418
  * s11d_dsmoke2_800CC4B8
  * s11d_dsmoke2_800CC794
  * s11d_dsmoke2_800CCAB4
- Keep implemented functions: 800CC08C, 800CC648, 800CC730, 800CCC60,
  800CCCE0, 800CCD54

Known Issue:
- func_80031298 address conflicts with GM_SetMap (0x80031244-0x800312D0)
  Needs investigation to determine correct overlay function address

mak1jk committed:
s11d: Fix func_80031298 → func_800C6544 incorrect address

The function was incorrectly labeled as func_80031298, which conflicts
with GM_SetMap in the main executable (0x80031244-0x800312D0).

Analysis:
- func_80031298 address falls inside GM_SetMap range
- All surrounding functions use 0x800Cxxxx addresses (s11d overlay)
- Assembly file s11d_rope_800C6544.s exists and matches usage pattern
- Function is called between func_800C650C and before func_800C4DE0
- Returns int value checked for != 0 in all three call sites

Correct function: func_800C6544 (s11d overlay function)
